pkg/flow/flow.go

```diff
@@ -106,6 +106,12 @@ type Options struct {
 	// The controller does not itself listen here, but some components
 	// need to know this to set the correct targets.
 	HTTPListenAddr string
+
+	// OnExportsChange is called when the exports of the controller change.
+	// Exports are controlled by "export" configuration blocks. If
+	// OnExportsChange is nil, export configuration blocks are not allowed in the
+	// loaded config file.
+	OnExportsChange func(exports map[string]any)
```

Comment:
What do we envision this function to be used for?

Reply:
See the commit which adds an example module.string component. It's used to propagate exports back out to the parent component/module.

Reply:
Ok, what I expected. Haven't gotten to that component yet.